src/stores/modelStore.ts

```diff
@@ -0,0 +1,141 @@
+import { api } from '@/scripts/api'
+import { defineStore } from 'pinia'
+import { reactive, ref, Ref } from 'vue'
+
+/** (Internal helper) finds a value in a metadata object from any of a list of keys. */
+function _findInMetadata(metadata: any, ...keys: string[]): string | null {
+  for (const key of keys) {
+    if (key in metadata) {
+      return metadata[key]
+    }
+    for (const k in metadata) {
+      if (k.endsWith(key)) {
+        return metadata[k]
+      }
+    }
+  }
+  return null
+}
+
+/** Defines and holds metadata for a model */
+export class ComfyModelDef {
+  /** Proper filename of the model */
+  name = ref('')
+  /** Directory containing the model, eg 'checkpoints' */
+  directory = ref('')
+  /** Title / display name of the model, sometimes same as the name but not always */
+  title = ref('')
+  /** Metadata: architecture ID for the model, such as 'stable-diffusion-xl-v1-base' */
+  architecture_id = ref('')
+  /** Metadata: author of the model */
+  author = ref('')
+  /** Metadata: resolution of the model, eg '1024x1024' */
+  resolution = ref('')
+  /** Metadata: description of the model */
+  description = ref('')
```

Comment:
Drop all these `ref`, as Vue's ref is deep, i.e. as long as we have Pinia manage a collection of ComfyModel, the nested object trees all become reactive.